DOC: build and upload 1.18 documentation to numpy.org/doc

One of the [last steps in the release](https://numpy.org/devdocs/dev/releasing.html?highlight=how%20release#build-and-archive-documentation) is to upload documentation to numpy.org/doc via the https://github.com/numpy/doc repo. This is new to 1.18.

It would be nice if someone else would work through the steps to make sure the directions are clear,  but I can do it if desired.

I've been working with the documentation a bit recently, so I decided to give the instructions linked above a try. Here's my feedback on the procedure:

> Build and archive documentation
> 
> Do:
> 
> cd doc/
> make dist

There is quite a bit of setup required before `make dist` can be used. Some of the requirements are listed higher up in [the same document](https://numpy.org/devdocs/dev/releasing.html#build-and-archive-documentation) while others more specific to building the docs can be [found elsewhere](https://docs.scipy.org/doc/numpy/dev/#building-docs). Here's what I had to do to get the docs building on a linux system with python3.8.1 (using `virtualenv` with `virtualenvwrapper` to help with environment management):

```bash
git clone https://github.com/numpy/numpy.git
cd numpy && git checkout v1.18.0
mkvirtualenv npy1.18-doc
pip install cython sphinx pandoc matplotlib scipy ipython
cd doc
git submodule update --init
make dist
```

Using this procedure I managed to get past the successful building of the html with sphinx, but I've hit an intractable error when it comes to building the `latex` docs:

```
Latexmk: Summary of warnings from last run of (pdf)latex:
  Latex failed to resolve 388 citation(s)
Latexmk: Errors, so I did not complete making targets
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'numpy-ref.log' for details
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs, or warnings treated as errors.
make[2]: *** [Makefile:33: numpy-ref.pdf] Error 12
make[2]: Leaving directory '/home/ross/numpy/doc/build/latex'
make[1]: *** [Makefile:108: real-dist] Error 2
make[1]: Leaving directory '/home/ross/numpy/doc'
make: *** [Makefile:104: build/dist.tar.gz] Error 2
```

The fatal error from `numpy-ref.log` is:

```
! TeX capacity exceeded, sorry [parameter stack size=10000].                    
\@inmathwarn #1->                                                               
                 \ifmmode \@latex@warning {Command \protect #1 invalid in ma... 
l.148312 ...ikipedia.org/wiki/Two’s\_complement}                                
                                                  ^^M                           
If you really absolutely need more capacity,                                    
you can ask a wizard to enlarge me.                                             
                                                                                
                                                                                
Here is how much of TeX's memory you used:                                      
 24083 strings out of 492167                                                    
 461100 string characters out of 6131557                                        
 825410 words of memory out of 5000000                                          
 22413 multiletter control sequences out of 15000+600000                        
 85961 words of font info for 136 fonts, out of 8000000 for 9000                
 1142 hyphenation exceptions out of 8191                                        
 5005i,26n,10001p,1538b,789s stack positions out of 20000i,500n,10000p,200000b,80000s
!  ==> Fatal error occurred, no output PDF file produced!
```

This seems to be a problem with how `texlive` is configured on my system. I didn't see anything in the release instructions about how the `TeX` build environment is supposed to be configured for the successful building of the `latex` docs, so this is where I stopped.

I hope this is helpful - if anyone has specific questions/comments I'm happy to address them.

--- a/doc/HOWTO_RELEASE.rst.txt
+++ b/doc/HOWTO_RELEASE.rst.txt
@@ -405,9 +405,9 @@ create an archive of the documentation in the numpy/doc repo::
     make merge-doc
     # Now edit the ``index.html`` file in the repo to reflect the new content,
     # and commit the changes
-    git -C dist/merge commit -a "Add documentation for <version>"
+    git -C build/merge commit -am "Add documentation for <version>"
     # Push to numpy/doc repo
-    git -C push
+    git -C build/merge push
 
 
 Update PyPI

